pkg/bench/bench_test.go

```diff
@@ -1420,3 +1428,81 @@ func BenchmarkNameResolution(b *testing.B) {
 		b.StopTimer()
 	})
 }
+
+func BenchmarkFuncExprTypeCheck(b *testing.B) {
+	skip.UnderShort(b)
+	defer log.Scope(b).Close(b)
+
+	s, db, kvDB := serverutils.StartServer(
+		b, base.TestServerArgs{UseDatabase: "defaultdb"})
+	defer s.Stopper().Stop(context.TODO())
+
+	sqlDB := sqlutils.MakeSQLRunner(db)
+	sqlDB.ExecMultiple(b,
+		`CREATE SCHEMA sc1`,
+		`CREATE SCHEMA sc2`,
+		`CREATE FUNCTION abs(val INT) RETURNS INT CALLED ON NULL INPUT LANGUAGE SQL AS $$ SELECT val $$`,
+		`CREATE FUNCTION sc1.udf(val INT) RETURNS INT CALLED ON NULL INPUT LANGUAGE SQL AS $$ SELECT val $$`,
+		`CREATE FUNCTION sc1.udf(val STRING) RETURNS STRING LANGUAGE SQL AS $$ SELECT val $$`,
+		`CREATE FUNCTION sc1.udf(val FLOAT) RETURNS FLOAT LANGUAGE SQL AS $$ SELECT val $$`,
+		`CREATE FUNCTION sc2.udf(val INT) RETURNS INT LANGUAGE SQL AS $$ SELECT val $$`,
+	)
+
+	ctx := context.Background()
+	execCfg := s.ExecutorConfig().(sql.ExecutorConfig)
+	p, cleanup := sql.NewInternalPlanner("type-check-benchmark",
+		kvDB.NewTxn(ctx, "type-check-benchmark-planner"),
+		username.RootUserName(),
+		&sql.MemoryMetrics{},
+		&execCfg,
+		sessiondatapb.SessionData{
+			Database: "defaultdb",
+		},
+	)
+
+	defer cleanup()
+	semaCtx := p.(sql.PlanHookState).SemaCtx()
+	sp := sessiondata.MakeSearchPath(append(sessiondata.DefaultSearchPath.GetPathArray(), "sc1", "sc2"))
+	semaCtx.SearchPath = &sp
+
+	testCases := []struct {
+		name    string
+		exprStr string
+	}{
+		{
+			name:    "builtin called on null input",
+			exprStr: "md5('some_string')",
+		},
+		{
+			name:    "builtin not called on null input",
+			exprStr: "parse_timetz('some_string')",
+		},
+		{
+			name:    "builtin aggregate",
+			exprStr: "corr(123, 321)",
+		},
+		{
+			name:    "udf same name as builtin",
+			exprStr: "abs(123)",
+		},
+		{
+			name:    "udf across different schemas",
+			exprStr: "udf(123)",
+		},
+	}
+
+	for _, tc := range testCases {
+		b.Run(tc.name, func(b *testing.B) {
+			expr, err := parser.ParseExpr(tc.exprStr)
+			require.NoError(b, err)
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				for j := 0; j < 1000; j++ {
```

Comment:
What is this `j` part?

Reply:
the [scripts/bench](https://github.com/cockroachdb/cockroach/blob/master/scripts/bench#L49-L49) script use `10` as benchmark count which seems low. I think I'm just lazy here and give it a `1000` run so the script can have some how reasonable result when running this benchmark.

Reply:
Ah, I see, let me remove that.